ibm/data_source_ibm_schematics_output.go

```diff
@@ -1,37 +1,77 @@
-// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
-// Licensed under the Mozilla Public License v2.0
+/**
+ * (C) Copyright IBM Corp. 2021.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 
 package ibm
 
 import (
 	"encoding/json"
 	"fmt"
+	"log"
+	"time"
 
+	"github.com/IBM/schematics-go-sdk/schematicsv1"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-func dataSourceSchematicsOut() *schema.Resource {
+func dataSourceIBMSchematicsOutput() *schema.Resource {
 	return &schema.Resource{
-		Read: resourceIBMSchematicsOutRead,
+		Read: dataSourceIBMSchematicsOutputRead,
 
 		Schema: map[string]*schema.Schema{
 			"workspace_id": {
 				Type:        schema.TypeString,
 				Required:    true,
-				Description: "The id of workspace",
+				Description: "The ID of the workspace for which you want to retrieve output values. To find the workspace ID, use the `GET /workspaces` API.",
 			},
 			"template_id": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
 				Description: "The id of template",
 			},
-			"type": {
-				Type:     schema.TypeInt,
-				Computed: true,
-			},
-			"output_values": {
-				Type:     schema.TypeMap,
-				Computed: true,
+			"output_values": &schema.Schema{
```

Comment:
This is due to the generated code from schema and the usage of the SDK to get the result. Changed back the output_values to TypeMap and handled the code accordingly so that the type is compatible with previous version